addons/mail/models/discuss/discuss_channel.py

```diff
@@ -59,6 +59,9 @@ def _generate_random_token(self):
         compute='_compute_channel_partner_ids', inverse='_inverse_channel_partner_ids',
         search='_search_channel_partner_ids')
     channel_member_ids = fields.One2many('discuss.channel.member', 'channel_id', string='Members')
+    parent_channel_id = fields.Many2one("discuss.channel", help="Parent channel", ondelete="cascade")
+    sub_channel_ids = fields.One2many("discuss.channel", "parent_channel_id", string="Sub Channels")
+    from_message_id = fields.Many2one("mail.message", help="The message the channel was created from.")
```

Comment:
How do we want to handle cascade of message?
We need to make sure `from_message_id` exists when using it in code.